### PR TITLE
Removed plain `assert`s and replaced them with ValidationUtil function

### PR DESCRIPTION
**Pull request by yfarjoun**
- some asserts remain, but they have been fixed in a different PR.

### Description

_Give your PR a **concise** yet **descriptive** title_
_Please explain the changes you made here._
_Explain the **motivation** for making this change. What existing problem does the pull request solve?_
_Mention any issues fixed, addressed or otherwise related to this pull request, including issue numbers or hard links for issues in other repos._
_You can delete these instructions once you have written your PR description._

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

**Comment by lbergelson** [comment]
Yes... asserts are evil.

**Comment by yfarjoun** [comment]
feel free to review this @lbergelson ....

File: src/main/java/picard/vcf/GenotypeConcordanceStates.java [hunk 1]
```diff
@@ -45,7 +47,7 @@
         public static TruthState getHom(final int alleleIdx) {
             if (alleleIdx == 0) return HOM_REF;
             if (alleleIdx == 1) return HOM_VAR1;
-            assert false;
+            ValidationUtils.validateArg(false,"Shouldn't be here.");
```
Comment (pshapiro4broad) [inline src/main/java/picard/vcf/GenotypeConcordanceStates.java]: In this case, why not `throw new PicardException("Invalid allele index: " + alleleIdx);`? This doesn't seem like a proper use of `validateArg()`.

File: src/main/java/picard/analysis/CollectWgsMetricsWithNonZeroCoverage.java [hunk 2]
```diff
@@ -186,7 +187,7 @@ protected WgsMetrics generateWgsMetrics(final IntervalList intervals,
 
     @Override
     protected WgsMetricsCollector getCollector(final int coverageCap, final IntervalList intervals) {
-        assert(coverageCap == this.collector.coverageCap);
+        ValidationUtils.validateArg(coverageCap == this.collector.coverageCap,()->"coverageCap has to be the same as the internal coverageCap, found " + coverageCap + " and " + this.collector.coverageCap);
```
Comment (pshapiro4broad) [inline src/main/java/picard/analysis/CollectWgsMetricsWithNonZeroCoverage.java]: Please reformat the changed lines in the IDE. In this case I would expect to see spaces around the `->` operator, and the line length seems long to me.

File: src/main/java/picard/fastq/BamToBfqWriter.java [hunk 3]
```diff
@@ -1,444 +1,446 @@
...
-import java.util.ArrayList;
-import java.util.Iterator;
-import java.util.List;
-
-/**
- * Class to take unmapped reads in BAM file format and create Maq binary fastq format file(s) --
- * one or two of them, depending on whether it's a paired-end read.  This relies on the unmapped
- * BAM file having all paired reads together in order.
- */
-public class BamToBfqWriter {
```
Comment (pshapiro4broad) [inline src/main/java/picard/fastq/BamToBfqWriter.java]: Did you mean to delete this class? This seems unrelated to your other changes, and maybe deserves its own PR.

File: src/test/java/picard/fingerprint/CrosscheckFingerprintsTest.java [hunk 4]
```diff
@@ -948,7 +949,7 @@ private void doTest(final String[] args, final File metrics, final int expectedR
 
             } catch (NoSuchFieldException e) {
                 e.printStackTrace();
-                assert false;
+                ValidationUtils.validateArg(false,"Shouldn't be here.");
```
Comment (lbergelson) [inline src/test/java/picard/fingerprint/CrosscheckFingerprintsTest.java]: just throw here

File: src/main/java/picard/vcf/GenotypeConcordanceStates.java [hunk 5]
```diff
@@ -56,7 +58,7 @@ public static TruthState getVar(final int allele0idx, final int allele1idx) {
             if (allele0idx == 1 && allele1idx == 2) return HET_VAR1_VAR2;
             if (allele0idx == 2 && allele1idx == 1) return HET_VAR1_VAR2;
 
-            assert false;
+            ValidationUtils.validateArg(false,"Shouldn't be here.");
```
Comment (lbergelson) [inline src/main/java/picard/vcf/GenotypeConcordanceStates.java]: Just throw if we shouldn't be here.

File: src/main/java/picard/fastq/BamToBfqWriter.java (hunk 3 above)
Comment (yfarjoun) [reply src/main/java/picard/fastq/BamToBfqWriter.java]: this seemed to have been due to the rebase...I didn't do it...

File: src/main/java/picard/fastq/BamToBfqWriter.java [hunk 6]
```diff
@@ -1,444 +1,446 @@
...
+
+        final char[] seqs = rec.getReadString().toCharArray();
+        final char[] quals = rec.getBaseQualityString().toCharArray();
+
+        int retainedLength = seqs.length;
+        if (clipAdapters){
+            // adjust to a shorter length iff clipping tag exists
+            Integer trimPoint = rec.getIntegerAttribute(ReservedTagConstants.XT);
+            if (trimPoint != null) {
+                ValidationUtils.validateArg(rec.getReadLength() == seqs.length, () -> "length of read and seqs differ. Found " + rec.getReadLength() + " and '" + seqs.length + ".");
```
Comment (yfarjoun) [inline src/main/java/picard/fastq/BamToBfqWriter.java]: this is the actual change in the file...everything else is due to newline differences.....

**Comment by yfarjoun** [comment]
since I added a commit...I cannot merge without another review....

File: src/main/java/picard/vcf/GenotypeConcordanceStates.java (hunk 1 above)
Comment (lbergelson) [reply src/main/java/picard/vcf/GenotypeConcordanceStates.java]: @yfarjoun Wait, I lied, you didn't respond to Phil here.

File: src/main/java/picard/vcf/GenotypeConcordanceStates.java (hunk 5 above)
Comment (lbergelson) [reply src/main/java/picard/vcf/GenotypeConcordanceStates.java]: Same

File: src/main/java/picard/vcf/GenotypeConcordanceStates.java [hunk 7]
```diff
@@ -109,8 +111,7 @@ public static CallState getHom(final int alleleIdx) {
             if (alleleIdx == 2) return HOM_VAR2;
             if (alleleIdx == 3) return HOM_VAR3;
 
-            assert false;
-            return null;
+            throw new IllegalStateException("Shouldn't be here.");
```
Comment (lbergelson) [inline src/main/java/picard/vcf/GenotypeConcordanceStates.java]: here

File: src/main/java/picard/vcf/GenotypeConcordanceStates.java [hunk 8]
```diff
@@ -124,22 +125,22 @@ public static CallState getHet(int allele0idx, int allele1idx) {
                 if (allele1idx == 1) return HET_REF_VAR1;
                 if (allele1idx == 2) return HET_REF_VAR2;
                 if (allele1idx == 3) return HET_REF_VAR3;
-                assert false;
+                ValidationUtils.validateArg(false,"Shouldn't be here.");
```
Comment (lbergelson) [inline src/main/java/picard/vcf/GenotypeConcordanceStates.java]: to

File: src/main/java/picard/vcf/GenotypeConcordanceStates.java [hunk 9]
```diff
@@ -124,22 +125,22 @@ public static CallState getHet(int allele0idx, int allele1idx) {
...
                 return null;
             }
 
             //HET CASES
             if(allele0idx == 1) {
                 if (allele1idx == 2) return HET_VAR1_VAR2;
                 if (allele1idx == 3) return HET_VAR1_VAR3;
-                assert false;
+                ValidationUtils.validateArg(false,"Shouldn't be here.");
```
Comment (lbergelson) [inline src/main/java/picard/vcf/GenotypeConcordanceStates.java]: and

File: src/main/java/picard/vcf/GenotypeConcordanceStates.java [hunk 10]
```diff
@@ -124,22 +125,22 @@ public static CallState getHet(int allele0idx, int allele1idx) {
...
                 return null;
             }
 
             if(allele0idx == 2 && allele1idx == 3) return HET_VAR3_VAR4; //special case not a mistake.
             if(allele0idx == 3 && allele1idx == 4) return HET_VAR3_VAR4;
 
-            assert false;
+            ValidationUtils.validateArg(false,"Shouldn't be here.");
```
Comment (lbergelson) [inline src/main/java/picard/vcf/GenotypeConcordanceStates.java]: here

**Comment by yfarjoun** [comment]
thanks @lbergelson fixed.

File: src/main/java/picard/vcf/GenotypeConcordanceStates.java [hunk 11]
```diff
@@ -1,5 +1,7 @@
 package picard.vcf;
 
+import htsjdk.utils.ValidationUtils;
```
Comment (pshapiro4broad) [inline src/main/java/picard/vcf/GenotypeConcordanceStates.java]: import is now unused?

File: src/test/java/picard/fingerprint/CrosscheckFingerprintsTest.java [hunk 12]
```diff
@@ -3,6 +3,7 @@
 import htsjdk.samtools.SamReader;
 import htsjdk.samtools.metrics.MetricsFile;
 import htsjdk.samtools.util.IOUtil;
+import htsjdk.utils.ValidationUtils;
```
Comment (pshapiro4broad) [inline src/test/java/picard/fingerprint/CrosscheckFingerprintsTest.java]: Also unused import
